content/en/blog/trust-chain/index.md

```diff
@@ -0,0 +1,362 @@
+authorization with JWT and authentication with FIDO2 regardless of which type of
+the entity needs to be authenticated: individuals, organizations, legal
+entities, or system components. For us, it implicated that we needed FIDO2 for
+service agents, which meant that a *headless* FIDO2 Authenticator was required.
+
+{{< imgproc cover-deployment.png Resize "1591x" >}}
+<em>All Key Components of The System Architecture</em>
+{{< /imgproc >}}
+
+Architectural requirements for the solution were quite complex because we wanted
+to have security covered, not to compromise performance, and still support
+polyglot development.
+
+## Polyglot Authenticator Interface
+
+FIDO2/WebAuthn specification gives a well over a description of how main
+components work. Here we focus on the two most important ones. The first is the
+authenticator registration flow which is presented picture below.
+
+![WebAuthn Registration](https://www.w3.org/TR/webauthn/images/webauthn-registration-flow-01.svg)
+<p align = "center"> FIDO2 Authenticator Registration - www.w3.org </p>
+
+To summarise, the above flow registers a new instance of an authenticator. Then
+it verifies that the same authenticator is bound to the account. That is done
+using a unique public/private key pair where the private key is in the
+authenticator. Note that the authenticator doesn't map a particular user to an
+account. That is done thru the other process flow and by the [relying
+party](https://www.w3.org/TR/webauthn/#webauthn-relying-party).
+
+The flow below shows how a registered authenticator is used to authenticate the
+account holder.
+
+![WebAuthn Authentication](https://www.w3.org/TR/webauthn/images/webauthn-authentication-flow-01.svg)
+<p align = "center"> FIDO2 Authentication - www.w3.org </p>
+
+The Command pattern was the perfect solution for the first implementation
+because it supported all of our use cases, but same time was simplest. Most
+straightforward to integration was naturally with a programming language it was
+implemented which was Go.
+
+The second thing was to figure out how we would like to implement interprocess
+communication. For that, the command pattern is suited very well. Fill the
+command with all the needed data and give one of the operations we were
+supporting: `register` and `login` from the FIDO2 standard. The process
+communication is handled just as the process starts by reading the command from
+JSON. That is suited for node.js use as well. (For the record, my fantastic
+colleague Laura did all the needed node.js work.)
+
+When we considered a security, we followed our post-compromise principle. We
+didn't (yet) try to solve the situation where someone managed to hack the server
+and hook a debugger to our processes without our notice. To solve that, we need
+[TEE](https://en.wikipedia.org/wiki/Trusted_execution_environment) or similar.
+Our specification is ready, but before the implementation, we should think if
+it's worth it.
```

Comment:
Maybe mention that the level of security (and the need for TEE) depends also from the severity of the use case?